drivers/pinctrl/pinctrl_xmc4xxx.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gpio/gpio_xmc4xxx.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/xmc4xxx-pinctrl.h>
+
+#include <xmc_gpio.h>
+
+static const struct device *const gpio_ports[] = {
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio0)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio1)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio2)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio3)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio4)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio5)),
+};
+
+static int pinctrl_configure_pin(const pinctrl_soc_pin_t pinmux)
+{
+	const struct device *gpio_device;
+	const struct gpio_xmc4xxx_config *gpio_config;
+	int port_id, pin, alt_fun, hwctrl;
+	XMC_GPIO_CONFIG_t pin_cfg = {0};
+
+	port_id = XMC4XXX_PINMUX_GET_PORT(pinmux);
+	if (port_id >= ARRAY_SIZE(gpio_ports)) {
+		return -EINVAL;
+	}
+
+	gpio_device = gpio_ports[port_id];
+	if (gpio_device == NULL || !device_is_ready(gpio_device)) {
```

Comment:
device_is_ready handles NULL input

Reply:
thanks, will update.
Do I even need to check that the gpio device has been initialized?
For example, this is checked in pinctrl_smt32.c, but not done in pinctrl_b91.c

Reply:
Yes, otherwise it's UB.